# HG changeset patch
# User Andreas Weber <andy.weber.aw@gmail.com>
# Date 1406756054 -7200
#      Wed Jul 30 23:34:14 2014 +0200
# Node ID bb2a03f9ed204e129861c50c746451f9cbf13bfb
# Parent  0257a936dc4a4d56dbb119ecebbafe27c2fc447d
Fix uigetdir, uigetfile, uiputfile. Remove check for nonexistent return value

* uigetfile.m: Don't check for existence of fltk script files. They are
  always distributed, even when built without FLTK. This function
  failed after moving the helper functions to ./private.

* uigetdir.m, uiputfile: same as uigetfile.m

* __is_function__.m: Remove comparison with impossible exist return value.

diff --git a/scripts/gui/private/__is_function__.m b/scripts/gui/private/__is_function__.m
--- a/scripts/gui/private/__is_function__.m
+++ b/scripts/gui/private/__is_function__.m
@@ -21,12 +21,12 @@
 ## Undocumented internal function.
 ## @end deftypefn
 
 ## Author: Michael Goffioul
 
 function result = __is_function__ (func)
 
   existval = exist (func);
-  result = (existval == 2 || existval == 3 || existval == 5 || existval == 6);
+  result = (existval == 2 || existval == 3 || existval == 5);
 
 endfunction
 
diff --git a/scripts/gui/uigetdir.m b/scripts/gui/uigetdir.m
--- a/scripts/gui/uigetdir.m
+++ b/scripts/gui/uigetdir.m
@@ -26,28 +26,22 @@
 ## @seealso{uigetfile, uiputfile}
 ## @end deftypefn
 
 ## Author: Kai Habel
 
 function dirname = uigetdir (init_path = pwd, dialog_name = "Select Directory to Open")
 
   if (! __octave_link_enabled__ ())
-    defaulttoolkit = get (0, "defaultfigure__graphics_toolkit__");
-    funcname = ["__uigetdir_", defaulttoolkit, "__"];
-    functype = exist (funcname);
-    if (! __is_function__ (funcname))
-      funcname = "__uigetdir_fltk__";
-      if (! __is_function__ (funcname))
-        error ("uigetdir: fltk graphics toolkit required");
-      elseif (! strcmp (defaulttoolkit, "gnuplot"))
-        warning ("uigetdir: no implementation for toolkit '%s', using 'fltk' instead",
-                 defaulttoolkit);
-      endif
+    tk = graphics_toolkit ();
+    funcname = ["uigetdir", tk, "__"];
+    if (numel (tk) > 0 && ! __is_function__ (funcname))
+      warning ("uigetdir: no implementation for toolkit '%s', using 'fltk' instead", tk);
     endif
+    funcname = "__uigetdir_fltk__";
   endif
 
   if (nargin > 2)
     print_usage ();
   endif
 
   if (!ischar (init_path) || !ischar (dialog_name))
     error ("uigetdir: INIT_PATH and DIALOG_NAME must be string arguments");
diff --git a/scripts/gui/uigetfile.m b/scripts/gui/uigetfile.m
--- a/scripts/gui/uigetfile.m
+++ b/scripts/gui/uigetfile.m
@@ -62,28 +62,22 @@
 ## @seealso{uiputfile, uigetdir}
 ## @end deftypefn
 
 ## Author: Kai Habel
 
 function [retfile, retpath, retindex] = uigetfile (varargin)
 
   if (! __octave_link_enabled__ ())
-    defaulttoolkit = get (0, "defaultfigure__graphics_toolkit__");
-    funcname = ["__uigetfile_", defaulttoolkit, "__"];
-    functype = exist (funcname);
-    if (! __is_function__ (funcname))
-      funcname = "__uigetfile_fltk__";
-      if (! __is_function__ (funcname))
-        error ("uigetfile: fltk graphics toolkit required");
-      elseif (! strcmp (defaulttoolkit, "gnuplot"))
-        warning ("uigetfile: no implementation for toolkit '%s', using 'fltk' instead",
-               defaulttoolkit);
-      endif
+    tk = graphics_toolkit ();
+    funcname = ["__uigetfile_", tk, "__"];
+    if (numel (tk) > 0 && ! __is_function__ (funcname))
+      warning ("uigetfile: no implementation for toolkit '%s', using 'fltk' instead", tk);
     endif
+    funcname = "__uigetfile_fltk__";
   endif
   
   if (nargin > 7)
     error ("uigetfile: number of input arguments must be less than eight");
   endif
 
   defaultvals = {cell(0, 2),         # File Filter
                  "Open File",        # Dialog Title
diff --git a/scripts/gui/uiputfile.m b/scripts/gui/uiputfile.m
--- a/scripts/gui/uiputfile.m
+++ b/scripts/gui/uiputfile.m
@@ -52,28 +52,22 @@
 ## @seealso{uigetfile, uigetdir}
 ## @end deftypefn
 
 ## Author: Kai Habel
 
 function [retfile, retpath, retindex] = uiputfile (varargin)
 
   if (! __octave_link_enabled__ ())
-    defaulttoolkit = get (0, "defaultfigure__graphics_toolkit__");
-    funcname = ["__uiputfile_", defaulttoolkit, "__"];
-    functype = exist (funcname);
-    if (! __is_function__ (funcname))
-      funcname = "__uiputfile_fltk__";
-      if (! __is_function__ (funcname))
-        error ("uiputfile: fltk graphics toolkit required");
-      elseif (! strcmp (defaulttoolkit, "gnuplot"))
-        warning ("uiputfile: no implementation for toolkit '%s', using 'fltk' instead",
-               defaulttoolkit);
-      endif
+    tk = graphics_toolkit ();
+    funcname = ["uiputfile", tk, "__"];
+    if (numel (tk) > 0 && ! __is_function__ (funcname))
+      warning ("uiputfile: no implementation for toolkit '%s', using 'fltk' instead", tk);
     endif
+    funcname = "__uiputfile_fltk__";
   endif
 
   if (nargin > 3)
     print_usage ();
   endif
 
   defaultvals = {cell(0, 2),     # File Filter
                  "Save File",    # Dialog Title
